src/schemas/components/SchemaFormPropertyInput.vue

```diff
@@ -29,17 +33,27 @@
     'update:value': [SchemaValue],
   }>()
 
+  const schema = useSchema()
+
   function update(value: unknown): void {
     emit('update:value', value)
   }
 
+  const property = computed(() => {
+    if (isPropertyWith(props.property, '$ref')) {
+      return merge({}, getSchemaDefinition(schema, props.property.$ref), props.property)
+    }
+
+    return props.property
+  })
```

Comment:
Here's where we're now making sure references are resolved. Used to happen in SchemaFormProperties.